app/models/piece.rb

```diff
@@ -2,5 +2,47 @@ class Piece < ActiveRecord::Base
   
   belongs_to :game
   belongs_to :player
+
+  def is_obstructed?(destination_x, destination_y)
+    location_x = self.x_position
+    location_y = self.y_position
+    if location_x == destination_x
+      # check for vertical obstruction
+      location_y > destination_y ? incrementer = -1 : incrementer = 1
+      position_y = location_y + incrementer
+      while position_y != destination_y
+        if game.pieces.where(x_position: location_x, y_position: position_y).any?
```

Comment:
Maybe possible to simplify by creating a range of your `position_y..destination_y`, then doing a `where(x_position: location_x, y_position: my_range)`